utils/CheckRequestAllowed.go

```diff
@@ -0,0 +1,95 @@
+package utils
+
+import (
+	"fmt"
+	"log"
+	"os"
+
+	"github.com/Real-Dev-Squad/feature-flag-backend/models"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/dynamodb"
+	"github.com/aws/aws-sdk-go/service/dynamodb/dynamodbattribute"
+	lambdaInvoke "github.com/aws/aws-sdk-go/service/lambda"
+)
+
+var raterLimiterFunctionName string
+var found bool
+
+func init() {
+	raterLimiterFunctionName, found = os.LookupEnv("RateLimiterFunction")
+	if !found {
+		log.Println("Rate limiter function name is not found")
+	}
+}
+
+func CheckRequestAllowed(db *dynamodb.DynamoDB, concurrencyValue int) {
+	// check the quota values
+	requestLimitInput := &dynamodb.GetItemInput{
+		TableName: aws.String("RequestLimit"),
+		Key: map[string]*dynamodb.AttributeValue{
+			"limitType": {
+				S: aws.String("pendingLimit"),
+			},
+		},
+	}
+
+	requestLimitResult, err := db.GetItem(requestLimitInput)
+	if err != nil {
+		log.Println(err, "is the error in request limit fetching")
+	}
+
+	requestLimitResponse := new(models.RequestLimit)
+	err = dynamodbattribute.UnmarshalMap(requestLimitResult.Item, requestLimitResponse)
+
+	if err != nil {
+		log.Println(err, "is the error")
+	}
+
+	if requestLimitResponse.LimitValue > 1 {
```

Comment:
lambdaInvoke.Invoke function even if there is an error updating the request limit counters. This could lead to the concurrency of all the other lambdas being set to zero, even if the request limit was not actually exceeded.